internal/vds/axis.h

```diff
@@ -0,0 +1,38 @@
+#ifndef AXIS_H
+#define AXIS_H
+
+#include <memory>
+#include <string>
+
+#include <OpenVDS/OpenVDS.h>
+
+#include "vds.h"
+
+class Axis {
+public:
+    Axis(
+        enum axis_name const                    axis_name,
+        OpenVDS::VolumeDataLayout const * const layout
+    );
+
+    int get_number_of_points() const noexcept(true);
+
+    int get_min() const noexcept(true);
+    int get_max() const noexcept(true);
+
+    std::string get_unit() const noexcept(true);
+
+    int get_voxel_index() const noexcept(true);
+    int get_index() const noexcept(true);
+
+    std::string get_name() const;
+
+    coordinate_system get_coordinate_system() const;
+private:
+    enum axis_name const axis_name;
+    int voxel_index;
+    int index;
```

Comment:
I agree. I guess more insecurity about how much `api` and `vds` may be in function and member names. 😄 

At the moment the class can also map between the meaning of an axis name used by a request to its corresponding axis in the VDS. This led to the two types of (badly named) indices in an effort to reduce "magic numbers".

The other review also expresses concern about the `index` and suggests removal. So maybe this will be solved then.